### DISCUSSION TITLE
AMQP protocol plan

### DISCUSSION THREAD

**Discussion by Liberxue** [apache/iggy#989] [Ideas]
@spetz hey Are there any plans to consider supporting the AMQP protocol?

**Comment by Liberxue** [comment]
I am willing to try adding provider support for this feature～～

**Comment by spetz** [comment]
Hey,
AMQP is a message queueing protocol while Iggy is a message stream, so it's not a typical message broker. Either the Iggy would have to support exchanges, queues, routing etc. (totally redesigned architecture), or we would need to come up with a custom specification for AMQP streaming (similar to RabbitMQ Streams for example).
